cpp/src/arrow/util/bit_stream_utils.h

```diff
@@ -263,8 +284,10 @@ inline void GetValue_(int num_bits, T* v, int max_bytes, const uint8_t* buffer,
 #pragma warning(disable : 4800 4805)
 #endif
     // Read bits of v that crossed into new buffered_values_
-    *v = *v | static_cast<T>(BitUtil::TrailingBits(*buffered_values, *bit_offset)
-                             << (num_bits - *bit_offset));
+    if (ARROW_PREDICT_TRUE(num_bits - *bit_offset < 64)) {
```

Comment:
please comment why this check is needed.

Reply:
To solve the runtime error in test job "AMD64 Ubuntu 20.04 C++ ASAN UBSAN":
/arrow/cpp/src/arrow/util/bit_stream_utils.h:289:30: runtime error: shift exponent 64 is too large for 64-bit type 'uint64_t' (aka 'unsigned long')
https://github.com/shanhuuang/arrow/runs/2986826720?check_suite_focus=true#logs
I met it when I try to read a file with only 3 records.

Reply:
thanks, makes esnse, could add a comment to the code.